### PR TITLE
Update pink nirvana and related documentation.

### PR DESCRIPTION
**Pull request by slontis**
I have seen multiple instances of references to the design document, and in a few cases copying of the pink nirvana diagram - so it was time to update it to match what we currently have. Also changed text related to the engine provider, since that was completely misleading.

### CODE REVIEW

File: docs/OpenSSL300Design.md [hunk 1]
```diff
@@ -201,20 +204,25 @@ here are as follows:
...
+    *   POST: Power On Self Test that perform:
+        *   KAT: Known Answer Tests
+        *   Integrity Check
     *   Low Level Implementations: This is the set of components that
         actually implement the cryptographic primitives (to meet the
         FIPS-mandated self-contained requirement).
-*   Engines Provider: A shim that allows existing engines to work
-    when called via the Core.
+*   ~~ Engines Provider: A shim that allows existing engines to work
+    when called via the Core. ~~
```
Comment (mattcaswell) [inline docs/OpenSSL300Design.md]: Maybe repeat the text "This was not implemented"...or just delete it

File: docs/OpenSSL300Design.md [hunk 2]
```diff
@@ -265,15 +273,25 @@ The physical packages new to this release are:
     there may be gaps/jumps in its version number since the previous
     release.
 
+*   ~~ Legacy module. This contains implementations of legacy
+    algorithms. ~~ This was not implemeted.
+
 *   Legacy module. This contains implementations of legacy
     algorithms.
```
Comment (mattcaswell) [inline docs/OpenSSL300Design.md:273]: It seems you intended to delete this old version of the text

File: docs/OpenSSL300Design.md [hunk 3]
```diff
@@ -265,15 +273,25 @@ The physical packages new to this release are:
...
 
-Engines will be built with a provider shim, to allow them to work as
+~~Engines will be built with a provider shim, to allow them to work as
 usual for cases when an ENGINE pointer is passed to some functions,
 and as providers when acting as default implementations. Engines
 compiled for pre-3.0.0 OpenSSL will need to be recompiled to work with
 OpenSSL 3.0.0. Details in
-[The ENGINE API](#the-engine-api) further down.
+[The ENGINE API](#the-engine-api) further down.~~
```
Comment (mattcaswell) [inline docs/OpenSSL300Design.md]: Perhaps preface this with "It was originally intended that Engines would have a provider shim..."

File: docs/OpenSSL300Design.md [hunk 4]
```diff
@@ -137,9 +140,9 @@ The architecture shall have the following features:
         libcrypto)
     b.  Legacy, with implementations of older algorithms (e.g, DES,
         MDC2, MD2, Blowfish, CAST)
-    c.  Engines, which include a compatibility layer between engines
+    c.  ~~Engines, which include a compatibility layer between engines
         designed for older versions of OpenSSL, and the new Provider
-        based approach.
+        based approach.~~ This was not implemented.
```
Comment (mattcaswell) [inline docs/OpenSSL300Design.md]: Should we just delete this line?

**Review by mattcaswell — CHANGES_REQUESTED** [review]
It the old design document actually linked anywhere on our website?

**Comment by slontis** [comment]
> It the old design document actually linked anywhere on our website?

Not that I can find.. Apart from the sitemap. It is getting found somehow.

**Comment by slontis** [comment]
Updated - mainly by deleting engine related comments.

File: docs/OpenSSL300Design.md [hunk 5]
```diff
@@ -201,20 +201,23 @@ here are as follows:
...
 *   Legacy Provider: Provides implementations of older algorithms that
     will be exposed via EVP-level APIs.
 *   3rd Party Providers: Eventually, third-parties may provide their
     own providers. A third-party provider, like any other provider,
     implements a set of algorithms that will be accessible to
     applications and other providers via the Core.
+*   Null Provider: A provider that does nothing. This can be useful
+    for testing that the correct library context is used.
+*   Base Provider: A provider for serialization of keys. The FIPS
+    provider requires this since it does not contains methods to
```
Comment (mattcaswell) [inline docs/OpenSSL300Design.md]: s/contains/contain/

File: docs/OpenSSL300Design.md [hunk 6]
```diff
@@ -201,20 +201,23 @@ here are as follows:
...
+    load keys. The Base provider is also embedded in the core.
```
Comment (mattcaswell) [inline docs/OpenSSL300Design.md]: "the core" may not be entirely clear. Do you mean "embedded in the default provider"?

**Review by mattcaswell — CHANGES_REQUESTED** [review]

**Review by mattcaswell — APPROVED** [review]

**Comment by romen** [comment]
This looks like a very good initiative to me, and would approve this, but I wanted to ask you first what do you think about the fact that in the 300_components and 300_packaging diagrams the 3rd party ENGINEs completely disappear: I am fine with it being that way in the `ToBe_*`` diagrams, as that is explained in the updated text, but in 3.0 the diagram does not show anymore how external ENGINEs fit in 3.0, where they still have some level of support.

**Comment by slontis** [comment]
I think it is sufficient to explain it in the document itself. Trying to draw it is probably not going to make it clear.

**Comment by levitte** [comment]
Merged

cc2c09421b4a3cf42ab8847afdf4baa75ce7e98b Update pink nirvana and related documentation.

**Pull request closed without merging**